python/paddle/fluid/layers/nn.py

```diff
@@ -5281,3 +5282,74 @@ def crop(x, shape=None, offsets=None, name=None):
         outputs={'Out': out},
         attrs=None if len(attrs) == 0 else attrs)
     return out
+
+
+def rank_loss(label, left, right, name=None):
+    """
+    **Rank loss layer for RankNet**
+
+    RankNet(http://icml.cc/2015/wp-content/uploads/2015/06/icml_ranking.pdf)
+    is a pairwise ranking model with one training sample consisting of a pair
+    of doc A and B, and the label P indicating that A is ranked higher than B
+    or not:
+    
+    P = {0, 1} or {0, 0.5, 1}, where 0.5 means no information about the rank of
+    the input pair.
+    
+    This layer takes three inputs: left (o_i), right (o_j) and label
+    (P_{i,j}), which represent the output score of RankNet for the two docs and
+    the label respectively, and yields the rank loss C_{i,j} using the following
+    equation:
+    
+    $$
+      C_{i,j} = -\tilde{P_{ij}} * o_{i,j} + \log(1 + e^{o_{i,j}}) \\
+      o_{i,j} =  o_i - o_j  \\
+      \tilde{P_{i,j}} = \left \{0, 0.5, 1 \right \} \ or \ \left \{0, 1 \right \}
+    $$
+    
+    This layer can take batch inputs with size batch_size (batch_size >= 1).
+    
+    Args:
+        label (Variable): The label indicating A ranked higher than B or not.
```

Comment:
label (Variable): indecates whether A is ranked higher than B or not.

Reply:
Done, thanks!